src/hotspot/share/compiler/compilationMemoryStatistic.cpp

```diff
@@ -186,10 +200,8 @@ class MemStatEntry : public CHeapObj<mtInternal> {
 
   // peak usage, bytes, over all arenas
   size_t _total;
-  // usage in node arena when total peaked
-  size_t _na_at_peak;
-  // usage in resource area when total peaked
-  size_t _ra_at_peak;
+  // usage per arena tag when total peaked
+  size_t _tags_size_at_peak[Arena::tag_count()];
```

Comment:
Can you please make sure Arena::tag_count() evaluates to constexpr? When in doubt, just use the enum value instead.

Reply:
Arena::tag_count() is declared as a constexpr. I wanted to avoid writing `static_cast<int>(Arena::Tag::tag_count)` every time I need tag_count, so I wrapped it in Arena::tag_count() and declared it with constexpr. Is that not sufficient to make it a constexpr?

Reply:
Okay then, that is fine.